doomworld_downloader/lmp_parser.py

```diff
@@ -8,3 +8,67 @@
     - Add a property to upload.ini that is points to the path of the parse_lmp script
   - Your output should be a dictionary of all relevant info for the final JSON
 """
+
+import logging
+import subprocess
+
+from .upload_config import CONFIG
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LMPData:
+    """Stores all uploader-relevant data for an LMP file
+
+    This is intended to be a very generic storage class that is mostly unaware of intricacies of the
+    LMP format like headers, footers, etc. That will be handled by the LMP library used underneath.
+    """
+    # Additional key notes:
+    #   Engine: Very high level info on what engine was used, either Doom or Boom
+    #   Version: Probably not very useful?
+    #   Episode: Always 1 for Doom 2
+    #   Play mode: Either "single / co-op" or "altdeath". In latter case, demo needs a note
+    #   Turbo: If yes, demo needs a note as a turbo run
```

Comment:
Not just a note, turbo would go into "other" for the category as well, right?

Reply:
I guess the data manager may need to know about "modifier" stuff. Like category reported as uv speed but turbo option activated -> not a "contradiction" but changes the result category :thinking:

Reply:
True, that's probably not true of just the data manager, though. I am wondering if it is sensible to have a preliminary class before the data manager that has some understanding of how the options play together. I am unsure how to make this happen any way other than very ugly conditionals and stuff like that 😅

Reply:
If other things need to check this too (like if we want a "tas flagger" object), I think we can just expose this from the manager and keep that stuff "in one box".

As the data manager gets bigger, I think the solution may be to split it a bit internally, like maybe inside it has an option manager with logic put in another object. Since those options are coming from the LMP analysis though, it could also make sense to generate that data here and pass it into the data manager as kind of a "bulk" report of data. Then the concerns are still isolated pretty well.

Reply:
Yeah, I guess it will end up being whatever is cleaner, heh